File: src/main/java/com/github/kindrat/cassandra/client/ui/window/editor/tables/TableButtons.java
package com.github.kindrat.cassandra.client.ui.window.editor.tables;

import com.github.kindrat.cassandra.client.util.UIUtil;
import javafx.geometry.HPos;
import javafx.geometry.VPos;
import javafx.scene.control.Button;
import javafx.scene.layout.*;

import static com.github.kindrat.cassandra.client.util.UIUtil.buildButton;
import static com.github.kindrat.cassandra.client.util.UIUtil.disable;
import static com.github.kindrat.cassandra.client.util.UIUtil.enable;

class TableButtons extends AnchorPane {

    private final Button plusButton;
    private final Button minusButton;
    private final Button commitButton;
    private final Button cancelButton;

    TableButtons() {
        setMaxHeight(40);
        setMinHeight(40);
        setPrefHeight(40);
        setMaxWidth(250);
        setMinWidth(250);
        setPrefWidth(250);

        GridPane gridPane = gridPane();
        getChildren().add(gridPane);
        UIUtil.fillParent(gridPane);

        plusButton = buildButton("+");
        gridPane.add(plusButton, 0, 0);

        minusButton = buildButton("-");
        gridPane.add(minusButton, 1, 0);

        commitButton = buildButton("✓");
        gridPane.add(commitButton, 3, 0);

        cancelButton = buildButton("∅");
        gridPane.add(cancelButton, 4, 0);
    }

    void enableButtons() {
        enable(plusButton, minusButton, commitButton, cancelButton);
    }

    void disableButtons() {
        disable(plusButton, minusButton, commitButton, cancelButton);
    }

    private GridPane gridPane() {
        GridPane gridPane = new GridPane();
        gridPane.getColumnConstraints().add(new ColumnConstraints(35, 35, 35, Priority.SOMETIMES, HPos.CENTER, true));
        gridPane.getColumnConstraints().add(new ColumnConstraints(35, 35, 35, Priority.SOMETIMES, HPos.CENTER, true));
        gridPane.getColumnConstraints().add(new ColumnConstraints(10, 100, 100, Priority.SOMETIMES, HPos.CENTER, true));
        gridPane.getColumnConstraints().add(new ColumnConstraints(35, 35, 35, Priority.SOMETIMES, HPos.CENTER, true));
        gridPane.getColumnConstraints().add(new ColumnConstraints(35, 35, 35, Priority.SOMETIMES, HPos.CENTER, true));
        gridPane.getColumnConstraints().add(new ColumnConstraints(10, 10, 10, Priority.SOMETIMES, HPos.CENTER, true));

        gridPane.getRowConstraints().add(new RowConstraints(30, 30, 30, Priority.SOMETIMES, VPos.CENTER, true));
        return gridPane;
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/window/editor/tables/TableButtons.kt
package com.github.kindrat.cassandra.client.ui.window.editor.tables

import com.github.kindrat.cassandra.client.util.UIUtil.buildButton
import com.github.kindrat.cassandra.client.util.UIUtil.disable
import com.github.kindrat.cassandra.client.util.UIUtil.enable
import com.github.kindrat.cassandra.client.util.UIUtil.fillParent
import javafx.event.EventHandler
import javafx.geometry.HPos
import javafx.geometry.VPos
import javafx.scene.control.Button
import javafx.scene.layout.*
import org.springframework.beans.factory.BeanFactory

internal class TableButtons : AnchorPane() {
    private val plusButton: Button
    private val commitButton: Button
    private val cancelButton: Button
    fun enableButtons() {
        enable(plusButton, commitButton, cancelButton)
    }

    fun disableButtons() {
        disable(plusButton, commitButton, cancelButton)
    }

    fun initActions(beanFactory: BeanFactory) {
        plusButton.onAction = EventHandler { beanFactory.getBean("tableEditor") }
    }

    private fun gridPane(): GridPane {
        val gridPane = GridPane()
        gridPane.columnConstraints.add(ColumnConstraints(35.0, 35.0, 35.0, Priority.SOMETIMES, HPos.CENTER, true))
        gridPane.columnConstraints.add(ColumnConstraints(35.0, 35.0, 35.0, Priority.SOMETIMES, HPos.CENTER, true))
        gridPane.columnConstraints.add(ColumnConstraints(10.0, 100.0, 100.0, Priority.SOMETIMES, HPos.CENTER, true))
        gridPane.columnConstraints.add(ColumnConstraints(35.0, 35.0, 35.0, Priority.SOMETIMES, HPos.CENTER, true))
        gridPane.columnConstraints.add(ColumnConstraints(35.0, 35.0, 35.0, Priority.SOMETIMES, HPos.CENTER, true))
        gridPane.columnConstraints.add(ColumnConstraints(10.0, 10.0, 10.0, Priority.SOMETIMES, HPos.CENTER, true))
        gridPane.rowConstraints.add(RowConstraints(30.0, 30.0, 30.0, Priority.SOMETIMES, VPos.CENTER, true))
        return gridPane
    }

    init {
        maxHeight = 40.0
        minHeight = 40.0
        prefHeight = 40.0
        maxWidth = 250.0
        minWidth = 250.0
        prefWidth = 250.0
        val gridPane = gridPane()
        children.add(gridPane)
        fillParent(gridPane)
        plusButton = buildButton("+")
        gridPane.add(plusButton, 0, 0)
        commitButton = buildButton("✓")
        gridPane.add(commitButton, 3, 0)
        cancelButton = buildButton("∅")
        gridPane.add(cancelButton, 4, 0)
    }
}